ocw/lib/azure.py

```diff
@@ -157,25 +161,52 @@ def cleanup_blob_containers(self) -> None:
                             self.log_info(f"Deleting blob {blob.name}")
                             self.container_client(container.name).delete_blob(blob.name, delete_snapshots="include")
 
-    def cleanup_images_from_rg(self) -> None:
-        self.log_dbg("Call cleanup_images_from_rg")
-        for item in self.list_images_by_resource_group(self.__resource_group):
+    def cleanup_images(self) -> None:
+        self.log_dbg("Call cleanup_images")
+        for item in self.list_images():
             if self.is_outdated(item.changed_time):
                 if self.dry_run:
                     self.log_info(f"Deletion of image {item.name} skipped due to dry run mode")
                 else:
                     self.log_info(f"Delete image '{item.name}'")
                     self.compute_mgmt_client().images.begin_delete(self.__resource_group, item.name)
 
-    def cleanup_disks_from_rg(self) -> None:
-        self.log_dbg("Call cleanup_disks_from_rg")
-        for item in self.list_disks_by_resource_group(self.__resource_group):
+    def cleanup_disks(self) -> None:
+        self.log_dbg("Call cleanup_disks")
+        for item in self.list_disks():
             if self.is_outdated(item.changed_time):
                 if self.compute_mgmt_client().disks.get(self.__resource_group, item.name).managed_by:
-                    self.log_warn(f"Disk is in use - unable delete {item.name}")
+                    self.log_warn(f"Disk is in use - skipping {item.name}")
                 else:
                     if self.dry_run:
                         self.log_info(f"Deletion of disk {item.name} skipped due to dry run mode")
                     else:
                         self.log_info(f"Delete disk '{item.name}'")
                         self.compute_mgmt_client().disks.begin_delete(self.__resource_group, item.name)
+
+    def cleanup_versions(self) -> None:
+        self.log_dbg("Call cleanup_versions")
+        try:
+            gallery = self.compute_mgmt_client().galleries.get(self.__resource_group, self.__gallery)
+        except ResourceNotFoundError:
+            self.log_err(f"No such gallery in resource group {self.__resource_group}: {self.__gallery}")
+            return
```

Comment:
this is something what not expected to happen so if it does I would like error to propagate further and cause mail send to us , so just remove this handler or replace return with raising same exception